errors.go

```diff
@@ -0,0 +1,527 @@
+
+	return errMsg.String()
+}
+
+// Message returns the message from the wrapped ClientMessage or an empty string
+func Message(err error) string {
+	cerr := &ClientMessage{}
+	if errors.As(err, &cerr) {
+		return cerr.Message()
+	}
+
+	return ""
+}
+
+// Messages returns a slice of messages from the ClientMessage's contained within the chain of errors
+func Messages(err error) []string {
+	cerr := &ClientMessage{}
+	if errors.As(err, &cerr) {
+		subMsgs := Messages(cerr.Unwrap())
+
+		msgs := make([]string, 0, len(subMsgs)+1)
+		msgs = append(msgs, cerr.Message())
+		msgs = append(msgs, subMsgs...)
+
+		return msgs
+	}
+
+	return nil
+}
+
+// CauseIsError returns true if the Cause of this error is an error vs a ClientMessage with nil error
+func CauseIsError(err error) bool {
+	cerr := &ClientMessage{}
+	if errors.As(err, &cerr) {
+		return CauseIsError(cerr.Unwrap())
+	}
+
+	return err != nil
+}
+
+// ClientMessage is a custom message type that can be used to return client messages
+type ClientMessage struct {
+	msgType       msgType
+	clientMessage string
+	error         error
+}
+
+// Message returns the client message
+func (c *ClientMessage) Message() string {
+	return c.clientMessage
+}
+
+// Error implements the error interface, returning the client message
+func (c *ClientMessage) Error() string {
+	return c.clientMessage
+}
+
+func (c *ClientMessage) Unwrap() error {
+	return c.error
+}
+
+func wrap(err error) errors.Chain {
+	return errors.WrapSkipFrames(err, "", 2)
+}
+
+// NewBadRequest creates a new empty client message with a BadRequest (400) return code
+func NewBadRequest() errors.Chain {
+	return wrap(&ClientMessage{
+		msgType: badRequest,
+	})
+}
+
+// NewUnauthorized creates a new empty client message with a Unauthorized (401) return code
+func NewUnauthorized() errors.Chain {
+	return wrap(&ClientMessage{
+		msgType: unauthorized,
+	})
+}
+
+// NewForbidden creates a new empty client message with a Forbidden (403) return code
+func NewForbidden() errors.Chain {
+	return wrap(&ClientMessage{
+		msgType: forbidden,
+	})
+}
+
+// NewNotFound creates a new empty client message with a NotFound (404) return code
+func NewNotFound() errors.Chain {
+	return wrap(&ClientMessage{
+		msgType: notFound,
+	})
+}
+
+// NewConflict creates a new empty client message with a Conflict (409) return code
+func NewConflict() errors.Chain {
+	return wrap(&ClientMessage{
+		msgType: conflict,
+	})
+}
+
+// NewInternalServerError creates a new empty client message with a InternalServerError (500) return code
+func NewInternalServerError() errors.Chain {
+	return wrap(&ClientMessage{
+		msgType: internalServerError,
+	})
+}
+
+// NewServiceUnavailable creates a new empty client message with a ServiceUnavailable (503) return code
+func NewServiceUnavailable() errors.Chain {
+	return wrap(&ClientMessage{
+		msgType: serviceUnavailable,
+	})
+}
+
+// NewBadRequestWithError creates a new empty client message with error and a BadRequest (400) return code
```

Comment:
It seems like these versions should say something along the lines of:
... wraps an existing error while creating an empty client message with a ____ (xyz) return code.

Reply:
Done